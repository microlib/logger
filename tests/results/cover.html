
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>multi: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/microlib/logger/pkg/multi/logger.go (87.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package multi

import (
        "fmt"
        "log"
)

const (
        INFO  = "info"
        DEBUG = "debug"
        WARN  = "warn"
        ERROR = "error"
        TRACE = "trace"
        PLAIN = "plain"
        HTML  = "html"
        COLOR = "color"
)

type colorRegistry struct {
        Red     string
        Green   string
        Blue    string
        Yellow  string
        Magento string
}

func NewLogger(mode, level string) *Logger <span class="cov8" title="1">{
        var cr *colorRegistry
        switch mode </span>{
        case "color":<span class="cov8" title="1">
                cr = &amp;colorRegistry{
                        Red:     "\x1b[1;31m [%s] \x1b[0m\t: %s",
                        Green:   "\x1b[1;36m [%s] \x1b[0m\t: %s",
                        Blue:    "\x1b[1;32m [%s] \x1b[0m\t: %s",
                        Yellow:  "\x1b[1;33m [%s] \x1b[0m\t: %s",
                        Magento: "\x1b[1;35m [%s] \x1b[0m\t: %s",
                }
                break</span>
        case "html":<span class="cov8" title="1">
                cr = &amp;colorRegistry{
                        Red:     "&lt;span style=\"color:#eb4034\"&gt;&amp;nbsp;[%s]&lt;/span&gt;&amp;nbsp;:&amp;nbsp;%s&lt;/br&gt;",
                        Green:   "&lt;span style=\"color:#34eb40\"&gt;&amp;nbsp;[%s]&lt;/span&gt;&amp;nbsp;:&amp;nbsp;%s&lt;/br&gt;",
                        Blue:    "&lt;span style=\"color:#3498eb\"&gt;&amp;nbsp;[%s]&lt;/span&gt;&amp;nbsp;:&amp;nbsp;%s&lt;/br&gt;",
                        Yellow:  "&lt;span style=\"color:#eba434\"&gt;&amp;nbsp;[%s]&lt;/span&gt;&amp;nbsp;:&amp;nbsp;%s&lt;/br&gt;",
                        Magento: "&lt;span style=\"color:#d234eb\"&gt;&amp;nbsp;[%s]&lt;/span&gt;&amp;nbsp;:&amp;nbsp;%s&lt;/br&gt;",
                }
                break</span>
        case "plain":<span class="cov8" title="1">
                cr = &amp;colorRegistry{
                        Red:     " [%s]\t: %s",
                        Green:   " [%s]\t: %s",
                        Blue:    " [%s]\t: %s",
                        Yellow:  " [%s]\t: %s",
                        Magento: " [%s]\t: %s",
                }
                break</span>
        }
        <span class="cov8" title="1">return &amp;Logger{Level: level, Colors: cr}</span>
}

type Logger struct {
        Level  string
        Colors *colorRegistry
}

func (logger Logger) Debug(message string) <span class="cov8" title="1">{
        if logger.Level == DEBUG || logger.Level == TRACE </span><span class="cov8" title="1">{
                log.Printf(logger.Colors.Green, "DEBUG", message)
        }</span>
}

func (logger Logger) Error(message string) <span class="cov8" title="1">{
        log.Printf(logger.Colors.Red, "ERROR", message)
}</span>

func (logger Logger) Warn(message string) <span class="cov0" title="0">{
        if logger.Level == WARN || logger.Level == DEBUG || logger.Level == INFO || logger.Level == TRACE </span><span class="cov0" title="0">{
                log.Printf(logger.Colors.Yellow, "WARN", message)
        }</span>
}

func (logger Logger) Trace(message string) <span class="cov8" title="1">{
        if logger.Level == TRACE </span><span class="cov8" title="1">{
                log.Printf(logger.Colors.Magento, "TRACE", message)
        }</span>
}

func (logger Logger) Info(message string) <span class="cov8" title="1">{
        if logger.Level == INFO || logger.Level == DEBUG || logger.Level == TRACE </span><span class="cov8" title="1">{
                log.Printf(logger.Colors.Blue, "INFO", message)
        }</span>
}

func (logger Logger) Errorf(msg string, val ...interface{}) <span class="cov8" title="1">{
        logger.Error(fmt.Sprintf(msg, val...))
}</span>

func (logger Logger) Debugf(msg string, val ...interface{}) <span class="cov8" title="1">{
        logger.Debug(fmt.Sprintf(msg, val...))
}</span>

func (logger Logger) Warnf(msg string, val ...interface{}) <span class="cov0" title="0">{
        logger.Warn(fmt.Sprintf(msg, val...))
}</span>

func (logger Logger) Tracef(msg string, val ...interface{}) <span class="cov8" title="1">{
        logger.Trace(fmt.Sprintf(msg, val...))
}</span>

func (logger Logger) Infof(msg string, val ...interface{}) <span class="cov8" title="1">{
        logger.Info(fmt.Sprintf(msg, val...))
}</span>
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
