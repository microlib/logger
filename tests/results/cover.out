mode: set
github.com/microlib/logger/pkg/multi/logger.go:27.44,29.14 2 1
github.com/microlib/logger/pkg/multi/logger.go:58.2,58.42 1 1
github.com/microlib/logger/pkg/multi/logger.go:30.15,38.8 2 1
github.com/microlib/logger/pkg/multi/logger.go:39.14,47.8 2 1
github.com/microlib/logger/pkg/multi/logger.go:48.15,56.8 2 1
github.com/microlib/logger/pkg/multi/logger.go:66.44,67.52 1 1
github.com/microlib/logger/pkg/multi/logger.go:67.52,69.3 1 1
github.com/microlib/logger/pkg/multi/logger.go:72.44,74.2 1 1
github.com/microlib/logger/pkg/multi/logger.go:76.43,77.100 1 0
github.com/microlib/logger/pkg/multi/logger.go:77.100,79.3 1 0
github.com/microlib/logger/pkg/multi/logger.go:82.44,83.27 1 1
github.com/microlib/logger/pkg/multi/logger.go:83.27,85.3 1 1
github.com/microlib/logger/pkg/multi/logger.go:88.43,89.76 1 1
github.com/microlib/logger/pkg/multi/logger.go:89.76,91.3 1 1
github.com/microlib/logger/pkg/multi/logger.go:94.61,96.2 1 1
github.com/microlib/logger/pkg/multi/logger.go:98.61,100.2 1 1
github.com/microlib/logger/pkg/multi/logger.go:102.60,104.2 1 0
github.com/microlib/logger/pkg/multi/logger.go:106.61,108.2 1 1
github.com/microlib/logger/pkg/multi/logger.go:110.60,112.2 1 1
